### PR TITLE
add GitHub links to style guide + other clarifications

### PR DESCRIPTION
**Pull request by taroface**

### CODE REVIEW

File: StyleGuide.md [hunk 1]
```diff
@@ -489,6 +490,14 @@ Links are marked with inline text surrounded by square brackets followed by the
     [--max-offset](#flags-max-offset)
     ```
 
+#### GitHub issues and pull requests
+
+[Release notes](https://www.cockroachlabs.com/docs/releases/index.html) and [technical advisories](https://www.cockroachlabs.com/docs/advisories/index.html) contain links to individual GitHub issues and pull requests.
+
+Issues and pull requests should be referenced by their corresponding number, prepended with `#`.
```
Comment (kathancox) [inline StyleGuide.md:497]: Really minor nit: You could switch up this sentence to make it active: 

You should reference issues and pull requests by their corresponding number, prepended with `#`.

OR 

Reference issues and pull requests by their corresponding number, prepended with `#`.

File: StyleGuide.md [hunk 2]
```diff
@@ -456,10 +456,11 @@ Whenever a CockroachDB feature is referenced, provide a link to the relevant doc
 
 Use Markdown reference-style links when several parts of the same page refer to the same target URL (e.g., [Release Notes](https://raw.githubusercontent.com/cockroachdb/docs/master/releases/v2.1.0-alpha.20180507.md)).
 
-Link capitalization can be either title or sentence case:
+Link capitalization should match our [capitalization rules](#capitalization-rules) for page titles and headers:
 
-- **Use title case** when referring to the linked doc by name (e.g., "See __Best Practices__ for more information").
-- **Use sentence case** - when linking in the middle of a sentence (e.g., "[…] follow the __identifier rules__ when creating […]").
+- **Use title case** when referring to the linked doc by its page title (e.g., "See __Best Practices__ for more information").
```
Comment (kathancox) [inline StyleGuide.md:461]: Really like this, super clear!

**Review by kathancox — APPROVED** [review]
One non-blocking nit. Very helpful addition for clarity!

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13933)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kathancox)

___
*[`StyleGuide.md` line 461 at r1](https://reviewable.io/reviews/cockroachdb/docs/13933#-N2rYYJsPI8D6XHG8cEr-r1-461:-N2rayNjA19h0AV0ePAx:b32vin) ([raw file](https://github.com/cockroachdb/docs/blob/79aae5c0b04c9f44ff4fffad395388885ec17568/StyleGuide.md#L461)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Really like this, super clear!
</blockquote></details>

:D
___
*[`StyleGuide.md` line 497 at r1](https://reviewable.io/reviews/cockroachdb/docs/13933#-N2rYYJsPI8D6XHG8cEr-r1-497:-N2rb0vn0vcOse3U2eZH:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/79aae5c0b04c9f44ff4fffad395388885ec17568/StyleGuide.md#L497)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Really minor nit: You could switch up this sentence to make it active: 

You should reference issues and pull requests by their corresponding number, prepended with `#`.

OR 

Reference issues and pull requests by their corresponding number, prepended with `#`.
</blockquote></details>

Done.

**Comment by taroface** [comment]
TFTR!

**Pull request merged**